src/worker/ghidra/ghidra.cc

```diff
@@ -0,0 +1,160 @@
+#include <sleigh/loadimage.hh>
+
+#include <iostream>
+#include <exception>
+#include <typeinfo>
+#include <stdexcept>
+
+#include "sleighMishegos.hh"
+#include "../worker.h"
+#include "mish_common.h"
+
+extern "C" {
+char *worker_name = (char *)"ghidra";
+
+void worker_ctor();
+
+void try_decode(decode_result *result, uint8_t *raw_insn, uint8_t length);
+}
+
+// This is a tiny LoadImage class which feeds the executable bytes to the translator
+class MyLoadImage : public LoadImage {
+  uintb baseaddr;
+  int4 length;
+  uint1 *data;
+
+public:
+  MyLoadImage(uintb ad, uint1 *ptr, int4 sz) : LoadImage("nofile") {
```

Comment:
Is `nofile` a special sentinel here? I'm not familiar with `LoadImage`; assuming that's part of Ghidra's APIs.

Reply:
Not special. It was used in the `sleighexample.cc` file here also https://github.com/NationalSecurityAgency/ghidra/blob/47f76c78d6b7d5c56a9256b0666620863805ff30/Ghidra/Features/Decompiler/src/decompile/cpp/sleighexample.cc#L72